test/840_weave_kube_3_test.sh

```diff
@@ -72,17 +73,38 @@ assert "run_on $HOST2 ps aux | grep -c '[d]efunct'" "0"
 assert "run_on $HOST3 ps aux | grep -c '[d]efunct'" "0"
 
 # See if we can get some pods running that connect to the network
-run_on $HOST1 "$KUBECTL run hello --image=weaveworks/hello-world --replicas=3"
+run_on $HOST1 "$KUBECTL run nettest --image=$IMAGE --replicas=3 -- -peers=3 -dns-name=nettest.default.svc.cluster.local."
+# Create a headless service so they can be found in Kubernetes DNS
+run_on $HOST1 "$KUBECTL create -f -" <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: nettest
+spec:
+  clusterIP: None
+  ports:
+  - port: 80
+    targetPort: 80
```

Comment:
Oh, it is actually [named this way in Kubernetes](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services). TIL.

Okay, so that's just for the labels/selectors and DNS as mentioned in the comment.
I think it could be valuable to add a comment saying the ports aren't used/do not need to match, but cannot be removed because Kubernetes complains otherwise.

With such a comment, I think it may be even better to keep a different port to show it doesn't matter / it not used, than to have them to the same value, but this is less important than the comment IMO.